test/parallel/test-http-send-file-fd.js

```diff
@@ -0,0 +1,39 @@
+'use strict';
+
+const common = require('../common');
+const http = require('http');
+const fixtures = require('../common/fixtures');
+const assert = require('assert');
+const fs = require('fs');
+
+const fname = fixtures.path('elipses.txt');
+
+const expected = fs.readFileSync(fname, 'utf8');
+
+const server = http.createServer(common.mustCall((req, res) => {
+  req.resume();
+  fs.open(fname, (err, fd) => {
+    assert(!err);
+    res.writeFile(fd, (err) => {
```

Comment:
This should also include tests verifying the function of `start` and `end` as well as input argument validation